catatonit.c

```diff
@@ -193,20 +194,46 @@ static int make_foreground(sigset_t *sigmask)
 	return 0;
 }
 
+/*
+ * If the LISTEN_PID environment variable is set to "1", rewrite it to
+ * point to the child process.
+ */
+static void rewrite_listen_pid_env(int parent_pid)
+{
+	char *listen_pid;
+	char pid_str[32];
+
+	sprintf(pid_str, "%d", parent_pid);
+
+	listen_pid = getenv("LISTEN_PID");
+	if (listen_pid != NULL && strcmp (listen_pid, pid_str) == 0) {
+		int r;
+
+		sprintf(pid_str, "%d", getpid());
+
+		r = setenv("LISTEN_PID", pid_str, 1);
+		if (r < 0)
+			warn("could not overwrite env variable LISTEN_PID: %m");
+	}
+}
+
 /*
  * Spawn a child process with the given arguments and signal map and make it a
  * faux-pid1 by placing it in the foreground. This is the main process which
  * catatonit is going to be managing throughout its life.
  */
 static int spawn_pid1(char *file, char **argv, sigset_t *sigmask)
 {
+	pid_t parent_pid = getpid();
```

Comment:
`getppid()` is simpler and doesn't require saving a variable. So `rewrite_listen_pid_env` doesn't really need to take an argument.